helm/prometheus-operator-app/templates/crd-install/kube-state-metrics-fix.yaml

```diff
@@ -0,0 +1,43 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ include "kube-prometheus-stack.ksmFix" . }}
+  namespace: {{ .Release.Namespace | quote }}
+  annotations:
+    # create hook dependencies in the right order
+    "helm.sh/hook-weight": "-1"
+    {{- include "kube-prometheus-stack.CRDInstallAnnotations" . | nindent 4 }}
+  labels:
+    app.kubernetes.io/component: {{ include "kube-prometheus-stack.ksmFix" . | quote }}
+    {{- include "kube-prometheus-stack.selectorLabels" . | nindent 4 }}
+    role: {{ include "kube-prometheus-stack.CRDInstallSelector" . | quote }}
+spec:
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/component: {{ include "kube-prometheus-stack.ksmFix" . | quote }}
+        {{- include "kube-prometheus-stack.selectorLabels" . | nindent 8 }}
+    spec:
+      serviceAccountName: {{ include "kube-prometheus-stack.crdInstall" . }}
+      securityContext:
+        runAsUser: 2000
+        runAsGroup: 2000
+      tolerations:
+      - key: node-role.kubernetes.io/master
+        effect: NoSchedule
+      containers:
+      - name: kubectl
+        image: "{{ .Values.image.registry }}/giantswarm/docker-kubectl:latest"
+        command:
+        - sh
+        - -c
+        - |
+          set -o errexit ; set -o xtrace ; set -o nounset
+
+          # piping stderr to stdout means kubectl's errors are surfaced
+          # in the pod's logs.
+          exec 2>&1
+
+          kubectl delete deployments.apps -l app.kubernetes.io/instance=prometheus-operator-app,app.kubernetes.io/name=kube-state-metrics --cascade=orphan
```

Comment:
Found out that this is the official fix, see https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus/README.md#to-150

Reply:
Nice